### DISCUSSION TITLE
Kestrel port binding default to 80 or 443 / No error/warn/info/debug when  "{abc}" provided

### DISCUSSION THREAD

**Discussion by tebeco** [dotnet/aspnetcore#24383] [General] [answered]
Hello,

After few long minutes (hours?) of debugging we found out that invalid port configuration will not say anything and silent.
Why did it takes so long ?
Because we were looking for the `https` binding that was `https://*:8443` on our POD, and it seems that another `appsettings...json` file still had an `http://*:{place.holder}` that was not replaced
So it seems that it defaulted to port `80` without saying anything at all.

The behavior was `AccessDeniedException` on a `Socket` (because Pods should not run as `root` so any binding bellow 1024 is forbidden)

The log level has been changed to debug for this capture :
![image](https://user-images.githubusercontent.com/2266487/88721795-08cffe80-d127-11ea-9393-09ac601f92f0.png)

so I wonder about 3 things now :
* should the `SocketException` / `AccessDenied` be explicit on the full binding url that was denied ? => `http://[::]:80` or` `http://*:80`
* should there be at least a `WARNING` on the fact that a string with letter and placerhoder is probably not a valid URI / Port ?
* is there any form of URI that would makes this valid: `http://*:{foo.bar.kix}` with the port `80` ?

**Comment by poke** [comment]
This appears to be a problem with [`BindingAdress.Parse`](https://github.com/dotnet/aspnetcore/blob/4aa5e03207a005d7310ba792c76f5339d74c3364/src/Http/Http/src/BindingAddress.cs#L78) which gets called early in the [`AddressBinder`](https://github.com/dotnet/aspnetcore/blob/4aa5e03207a005d7310ba792c76f5339d74c3364/src/Servers/Kestrel/Core/src/Internal/AddressBinder.cs#L104). Since the binder is not able to produce a better result from the input, it will eventually fall back to just binding on the port that the `BindingAdress` contains:

https://github.com/dotnet/aspnetcore/blob/4aa5e03207a005d7310ba792c76f5339d74c3364/src/Servers/Kestrel/Core/src/Internal/AddressBinder.cs#L135-L139

Unfortunately, `BindingAddress.Parse` does not appear to parse the input very well:

![BindingAddress.Parse producing a port 443](https://user-images.githubusercontent.com/132240/88723658-898ffa00-d129-11ea-94a1-688fc31fd8f3.png)

Calling `BindingAddress.Parse("https://*:{abc}")` will recognize `*:{abc}` as the host and assume [the default HTTPS port 443](https://github.com/dotnet/aspnetcore/blob/4aa5e03207a005d7310ba792c76f5339d74c3364/src/Http/Http/src/BindingAddress.cs#L147-L150).

**Comment by tebeco** [comment, marked as answer]
well then, created an issue on that one :
https://github.com/dotnet/aspnetcore/issues/24384
